types/wlr_output.c

```diff
@@ -267,6 +267,15 @@ void wlr_output_set_subpixel(struct wlr_output *output,
 	wlr_output_schedule_done(output);
 }
 
+void wlr_output_set_format(struct wlr_output *output,
```

Comment:
This seems to be a no-op.

Reply:
Right, so my understanding of the other output setting functions was to skip all of the `output->pending` and state setting if the output setting had not changed. Right now `wlr_output_set_format()` doesn't have any of the `output->pending` or state change stuff, and I wasn't sure if any of it was necessary to do things safely. In `wlr_output_enable_adaptive_sync()`, both state changing and `output->pending` shows up there, while `wlr_output_set_subpixel()` is doing something different.

Reply:
`wlr_output_set_subpixel` doesn't need to interact with the backend, so doesn't need to use `output->pending`.

`wlr_output_set_format` should make the backend de-allocate and re-allocate a buffer with the correct format right? This doesn't seem to be the case right now, `output->format` is never used.